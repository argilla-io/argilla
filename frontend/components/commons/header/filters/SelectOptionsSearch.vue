<template>
  <div class="options-search">
    <input
      v-model="searchText"
      class="options-search__input"
      type="text"
      autofocus
      :placeholder="placeholder"
    />
    <a href="#" class="options-search__clear" @click.prevent="clear">
      <svgicon
        v-if="allowClear && !!searchText"
        name="close"
        width="10"
        height="10"
        color="#9b9b9b"
      ></svgicon>
    </a>
  </div>
</template>

<script>
export default {
  props: {
    placeholder: {
      type: String,
      default: "Search...",
    },
    value: {
      type: String,
      default: "",
    },
    allowClear: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    searchText: {
      get() {
        return this.value;
      },
      set(value) {
        this.$emit("input", value);
      },
    },
  },
  methods: {
    clear() {
      this.$emit("clear");
    },
  },
};
</script>

<style scoped lang="scss">
.options-search {
  &__input {
    border: none;
    outline: none;
    height: 40px;
    background: transparent;
    width: calc(100% - 20px);
    @include input-placeholder {
      color: $black-37;
    }
  }
  &__clear {
    outline: none;
  }
}
</style>
