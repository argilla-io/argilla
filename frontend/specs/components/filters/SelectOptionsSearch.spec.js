import { mount } from "@vue/test-utils";
import SelectOptionsSearch from "@/components/commons/header/filters/SelectOptionsSearch";

function mountSelectOptionsSearch() {
  return mount(SelectOptionsSearch, {
    propsData: {
      value: "Search Input",
      allowClear: true,
    },
  });
}

describe("SelectOptionsSearch", () => {
  const spy = jest.spyOn(console, "error");
  afterEach(() => spy.mockReset());

  test("renders properly", () => {
    const wrapper = mountSelectOptionsSearch();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
