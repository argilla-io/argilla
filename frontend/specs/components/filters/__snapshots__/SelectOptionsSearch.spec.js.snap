// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectOptionsSearch renders properly 1`] = `<div class="options-search"><input type="text" autofocus="autofocus" placeholder="Search..." class="options-search__input"> <a href="#" class="options-search__clear"><svg version="1.1" viewBox="0 0 10 10" class="svg-icon svg-fill" style="width: 10px; height: 10px;"></svg></a></div>`;
